xskillscore/tests/test_mask_skipna.py

```diff
@@ -2,104 +2,89 @@
 def b():
-    time = pd.date_range("1/1/2000", "1/3/2000", freq="D")
+    time = pd.date_range('1/1/2000', '1/3/2000', freq='D')
     lats = np.arange(4)
     lons = np.arange(5)
     data = np.random.rand(len(time), len(lats), len(lons))
     da = xr.DataArray(
-        data, coords=[time, lats, lons], dims=["time", "lat", "lon"]
+        data, coords=[time, lats, lons], dims=['time', 'lat', 'lon']
     )
     return da
 
 
-def mask_data(da, dim):
+def mask_land_data(da):
     """
-    Masks sample data arbitrarily over specified dim.
+    Masks sample data arbitrarily like a block of land.
     """
-    # Mask an arbitrary region with NaNs (like a block of land).
-    return da.where(da[dim] > da[dim].isel({dim: 0}))
-
-
-@pytest.mark.parametrize("dim", AXES)
-def test_pearson_r_masked(a, b, dim):
-    a_masked = mask_data(a, dim[0])
-    b_masked = mask_data(b, dim[0])
-
-    res_skipna = pearson_r(a_masked, b_masked, dim, skipna=True)
-    res_no_skipna = pearson_r(a_masked, b_masked, dim, skipna=False)
-    assert np.isnan(res_no_skipna).all()
-    assert not np.isnan(res_skipna).any()
-
-
-@pytest.mark.parametrize("dim", AXES)
-def test_pearson_r_p_value_masked(a, b, dim):
-    a_masked = mask_data(a, dim[0])
-    b_masked = mask_data(b, dim[0])
-
-    res_skipna = pearson_r_p_value(a_masked, b_masked, dim, skipna=True)
-    res_no_skipna = pearson_r_p_value(a_masked, b_masked, dim, skipna=False)
-    # p-value defaults to exactly 1.0 instead of NaNs.
-    assert (res_no_skipna == 1.0).all()
-    assert not np.isnan(res_skipna).any()
-
-
-@pytest.mark.parametrize("dim", AXES)
-def test_rmse_masked(a, b, dim):
-    a_masked = mask_data(a, dim[0])
-    b_masked = mask_data(b, dim[0])
-
-    res_skipna = rmse(a_masked, b_masked, dim, skipna=True)
-    res_no_skipna = rmse(a_masked, b_masked, dim, skipna=False)
-    assert np.isnan(res_no_skipna).all()
-    assert not np.isnan(res_skipna).any()
-
-
-@pytest.mark.parametrize("dim", AXES)
-def test_mse_masked(a, b, dim):
-    a_masked = mask_data(a, dim[0])
-    b_masked = mask_data(b, dim[0])
-
-    res_skipna = mse(a_masked, b_masked, dim, skipna=True)
-    res_no_skipna = mse(a_masked, b_masked, dim, skipna=False)
-    assert np.isnan(res_no_skipna).all()
-    assert not np.isnan(res_skipna).any()
-
+    da.data[:, 1:3, 1:3] = np.nan
+    return da
 
-@pytest.mark.parametrize("dim", AXES)
-def test_mae_masked(a, b, dim):
-    a_masked = mask_data(a, dim[0])
-    b_masked = mask_data(b, dim[0])
 
-    res_skipna = mae(a_masked, b_masked, dim, skipna=True)
-    res_no_skipna = mae(a_masked, b_masked, dim, skipna=False)
-    assert np.isnan(res_no_skipna).all()
-    assert not np.isnan(res_skipna).any()
+@pytest.mark.parametrize('metric', correlation_metrics + distance_metrics)
+@pytest.mark.parametrize('dim', AXES)
+def test_metrics_masked(a, b, dim, metric):
+    """Test for all distance-based metrics whether result if skipna do not
+     contain any nans when applied along dim with nans."""
```

Comment:
some typos """Test for all distance-based metrics whether result of skipna does not
     contain any nans when applied along dim with nans."""